[#3603] Use different action + status texts for zaken/formulieren

pi-sigma opened the pull request:
## Summary

- 'Zaken' and 'Openstaande formulieren' receive different texts for the status as well as the action button/link. Also introduced an Enum to handle the case type ('Zaak' vs 'OpenSubmission').

- Refactored some classes, methods etc. to use Dutch terms referencing the domain (`submission` -> `formulier`, `SubmissionWithApiGroup` -> `FormulierWithApiGroup` etc.)

## Issue References

<!-- Link to related Taiga stories/issues: remove if not applicable -->

- Taiga User Story: https://taiga.maykinmedia.nl/project/open-inwoner/us/3603

## Checklist

- [x] CHANGELOG.rst updated

pi-sigma committed:
feat: use different action + status texts for zaken/formulieren

'Zaken' and 'Openstaande formulieren' receive different texts for the
status as well as the action button/link. Also introduced an Enum
to handle the case type ('Zaak' vs 'OpenSubmission').

swrichards commented on src/open_inwoner/openzaak/api_models.py:
```diff
@@ -142,7 +148,7 @@ def process_data(self) -> dict:
             "zaaktype_config": getattr(self, "zaaktype_config", None),
             "statustype_config": getattr(self, "statustype_config", None),
-            "case_type": "Zaak",
+            "case_type": TypeAanvraag.ZAAK.value,
```
Wouldn't it make more sense to do this at the level of the service? We enrich the core objects with other fields too there, e.g. the originating API group.

pi-sigma replied:
Makes sense, didn't think of this option.

swrichards reviewed (commented):
I think this is a useful addition, I'd just add the enum at a higher level: in the case list service.

pi-sigma committed:
refactor: rename API model 'OpenSubmission' to 'Formulier'

The API model for 'formulier aanvragen' is renamed from 'OpenSubmission'
to 'Formulier'. For conformity, we also rename the
'SubmissionWithApiGroup' dataclass to 'FormulierWithApiGroup' and use
Dutch terms for the 'CaseFilterFormOption' fields.

pi-sigma committed:
refactor: replace 'submission' with 'formulier' in ZGW servies and clients

In accordance with the renaming of the API model from 'OpenSubmission'
to 'Formulier', the ZGW 'FormClient' is renamed to 'FormulierenClient',
and the relevant methods and calls are renamed accordingly (e.g.
'fetch_submissions' -> 'fetch_formulieren').

swrichards commented on CHANGELOG.rst:
```diff
@@ -93,12 +93,14 @@ Nieuwe features
   Standaard blijven de hints ingeschakeld om het huidige gedrag te behouden.
 * [:taiga-us:`3580`, :pr:`2033`]: De zoekfunctie kan nu worden uitgeschakeld in de Algemene Configuratie.
-* [:taiga-is`3600`, :pr:`2046`]: 'Signing requets' toegevoegd aan admin. Beheerders kunnen nu een CSR
+* [:taiga-is:`3600`, :pr:`2046`]: 'Signing requets' toegevoegd aan admin. Beheerders kunnen nu een CSR
```
Also: 
```suggestion
* [:taiga-is:`3600`, :pr:`2046`]: 'Signing requests' toegevoegd aan admin. Beheerders kunnen nu een CSR
```
